### ISSUE TITLE
[REVIEW]: widgyts: Custom Jupyter Widgets for Interactive Data Exploration with yt

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1774]
**Submitting author:** @munkm (<a href="http://orcid.org/0000-0003-0117-5366">Madicken Munk</a>)
**Repository:** <a href="https://github.com/data-exp-lab/widgyts" target ="_blank">https://github.com/data-exp-lab/widgyts</a>
**Version:**  v0.3.2
**Editor:** @lpantano
**Reviewer:** @harpolea , @KayleighRutherford
**Archive:** <a href="https://doi.org/10.5281/zenodo.3583416" target="_blank">10.5281/zenodo.3583416</a>

## Status

[![status](https://joss.theoj.org/papers/f86e07ce58fe8bb24d928943663d2751/status.svg)](https://joss.theoj.org/papers/f86e07ce58fe8bb24d928943663d2751)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/f86e07ce58fe8bb24d928943663d2751"><img src="https://joss.theoj.org/papers/f86e07ce58fe8bb24d928943663d2751/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/f86e07ce58fe8bb24d928943663d2751/status.svg)](https://joss.theoj.org/papers/f86e07ce58fe8bb24d928943663d2751)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/data-exp-lab/widgyts" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@harpolea  & @KayleighRutherford, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @lpantano know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/data-exp-lab/widgyts">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@munkm) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/data-exp-lab/widgyts">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@munkm) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @harpolea , @KayleighRutherford it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01774/joss.01774/10.21105.joss.01774.pdf)

**Comment by lpantano** [comment]
:wave @KayleighRutherford, @harpolea, how is the review process going, do you think you are close to finish?

**Comment by harpolea** [comment]
I'm so sorry for the delay in finishing this! 

Overall, this is a really cool package. I've read over the documentation and played around with the examples notebooks, and everything there looks good to me. The python test suite has good coverage (though the coverage reports on `codecov` don't seem to be working at the moment?).

I have a couple of small suggestions for the paper before I tick off the last couple of points in the checklist, which I've listed in this issue: https://github.com/data-exp-lab/widgyts/issues/45

**Comment by lpantano** [comment]
@KayleighRutherford do you think you can finish the review soon?

**Comment by KayleighRutherford** [comment]
Hello, I did complete the checklist a while back - is there something more
I need to do/how do I close the issue?

**Comment by lpantano** [comment]
thanks for the reply @KayleighRutherford , I see some of the items not checked under your name checklist, can you double check it is complete, maybe it didn't sync when you did it. Thanks!

**Comment by KayleighRutherford** [comment]
Hi @lpantano, I've completed this and wrote to the authors to address the remaining issues, but am now unable to edit my checklist? Can you advise on how to fix this (I am logged in and obviously accepted the invite as outlined but have since lost the ability to edit for some reason). Please let me know and I'll close this out. Thanks for waiting on me! Sorry this took some time

**Comment by lpantano** [comment]
Thanks!, don't worry about the checklist, I can wait for  @munkm to make changes, and you will need to agree on those. Then I can accept the paper. Is there an issue for the changes? or was it an email? @munkm, just tell me when you are done with the changes.

**Comment by KayleighRutherford** [comment]
my comments are in issue #45 in the repo (sorry just realised I was supposed to create a new one..)

**Comment by munkm** [comment]
@lpantano @KayleighRutherford and @harpolea -- thank you all for the reviews (and helping shepherd the paper). I've opened some PRs hopefully addressing the reviewer comments in the repository (data-exp-lab/widgyts#46 and data-exp-lab/widgyts#47). After they get reviewed and merged, I think things should be good!

**Comment by munkm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1774 with the following error: 

 Error producing PDF.
! Missing $ inserted.
<inserted text> 
                $
l.404 ... data}} < n*[t_{\text{image_calc, server}

Looks like we failed to compile the PDF

**Comment by munkm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01774/joss.01774/10.21105.joss.01774.pdf)

**Comment by munkm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01774/joss.01774/10.21105.joss.01774.pdf)

**Comment by lpantano** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1088/0067-0049/192/1/9 is OK
- 10.1109/MCSE.2007.53 is OK

MISSING DOIs

- https://doi.org/10.1109/mcse.2007.55 may be missing for title: Matplotlib: A 2D graphics environment

INVALID DOIs

- None
```

**Comment by lpantano** [comment]
Hi @munkm, Any chance you can check the missing DOIs the test is finding?

**Comment by munkm** [comment]
Hi @lpantano I just pushed a change that should fix the doi in that reference. Should I regenerate the PDF?

**Comment by lpantano** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01774/joss.01774/10.21105.joss.01774.pdf)

**Comment by lpantano** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1109/MCSE.2007.55 is OK
- 10.1088/0067-0049/192/1/9 is OK
- 10.1109/MCSE.2007.53 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by lpantano** [comment]
@KayleighRutherford can you confirm that all your questions are addressed and checked the items if yes, or ask for more if no? Thanks

**Comment by KayleighRutherford** [comment]
@lpantano, yes I think everything has been addressed - looks good to me

**Comment by lpantano** [comment]
Thank you! @munkm, could you create a zenodo archive from the repo. The title and authors should match the paper information. After that I can add this to the archive link and we will be almost done!

**Comment by munkm** [comment]
Hi @lpantano I've created a release and made a zenodo archive from the repo based on the most current tagged release of the software. Should I make a different one that matches the paper title?

**Comment by lpantano** [comment]
I think you could change the title and author manually from that same zenodo? but if not we need one that matches the title and author.

**Comment by munkm** [comment]
Ah! Thanks for the tip @lpantano. I manually updated the name and author list on Zenodo. I think it should match now.

**Comment by lpantano** [comment]
Thanks! glad it worked, @munkm can you put the link here?

**Comment by munkm** [comment]
yep! here it is! 

https://zenodo.org/record/3583416

**Comment by lpantano** [comment]
@whedon set 10.5281/zenodo.3583416 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3583416" target="_blank">10.5281/zenodo.3583416</a> is the archive.

**Comment by lpantano** [comment]
@whedon set  v0.3.2 as version

**Comment by whedon** [comment]
OK.  v0.3.2 is the version.

**Comment by lpantano** [comment]
Hi @openjournals/joss-eics, I think you can check this one now. happy new year!

**Comment by kthyng** [comment]
I can jump in here!

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01774/joss.01774/10.21105.joss.01774.pdf)

**Comment by kthyng** [comment]
Hi @munkm! I am going to finish up the processing of your paper.

I am going through your paper now. Can you look into the reference for Ipywidgets? Should the final author be Min RK or should it be his full name?

**Comment by kthyng** [comment]
I see there is an issue with the references in the 2nd paragraph under "motivation" but I'm having trouble figuring out the source. For one thing, is there no author available for your "jupyter-matplotlib_2019" reference, nor "itkwidgets_2019", "bokeh_2019", and "bqplot_2019"?

**Comment by munkm** [comment]
Hi @kthyng -- thank you for catching these! Min's name should indeed be longer. I'll update it and push again. 

For the references in the 2nd paragraph, I used Zotero to import citation data on github repositories. None of these have a citation file, but there is a "publisher" information (e.g. Matplotlib developers) that was imported that could be used as authorship. When I use citeas it looks like this is what's used in place of individual authorship (e.g. https://citeas.org/cite/https://github.com/bokeh/bokeh). 

Which would be better in this case? Duplicating the publisher information to the authorship information? Or copying first few contributor names in the repo to the author section of the citation? I'll update with whichever you think is best.

**Comment by kthyng** [comment]
@munkm Whoa I didn't "citeas" existed! Very cool! I think if a project hasn't given a preferred citation and doesn't have a clear paper to cite, using the "citeas" approach of the publisher as author will probably look best. Can you try that out and see how it looks inline? You can test here if you want: https://whedon.theoj.org/

**Comment by munkm** [comment]
Hi @kthyng! I just heard about `citeas` recently, too! I'm a big fan! ❤️ 

Thank you for linking the paper preview service! It's so helpful. 

I've fixed the references based on opening issues in a few repositories to check what the package developers preferred. I think I've fixed most of the weird citation issues now and they look good in the preview service to me. Can you let me know if everything is ok on your end?

**Comment by munkm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01774/joss.01774/10.21105.joss.01774.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@kthyng are you still processing this one or do I help here too this week?

**Comment by kthyng** [comment]
@Kevin-Mattheus-Moerman Yes, sorry for my delay. I hope to get to it this afternoon.

**Comment by kthyng** [comment]
Hi @munkm! I just read through your paper, and have just a couple of typos in this [PR](https://github.com/data-exp-lab/widgyts/pull/51)

**Comment by munkm** [comment]
Thanks for catching them 🙏, it's really appreciated! Your fixes have been merged.

**Comment by kthyng** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1109/MCSE.2007.55 is OK
- 10.1088/0067-0049/192/1/9 is OK
- 10.1109/MCSE.2007.53 is OK
- 10.5281/zenodo.3351620 is OK
- 10.5281/zenodo.3603359 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kthyng** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1109/MCSE.2007.55 is OK
- 10.1088/0067-0049/192/1/9 is OK
- 10.1109/MCSE.2007.53 is OK
- 10.5281/zenodo.3351620 is OK
- 10.5281/zenodo.3603359 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1256

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1256, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
Hi all! This submission is almost wrapped up from my end, but I just perused the checklists and other items above and I see that @KayleighRutherford gave approval (though wasn't able to check the final items off due to permissions) but I'm not sure we heard a final word from @harpolea, particularly given an open issue [#45](https://github.com/data-exp-lab/widgyts/issues/45). I'd like to make sure this issue is satisfactorily completed before finalizing this submission.

**Comment by harpolea** [comment]
I'm happy to sign off on this - the modifications to the paper have addressed the suggestions I made, so all looks good to me!

**Comment by kthyng** [comment]
@harpolea Thanks!

I think we can finish then.

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1222625770868178944) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1259
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01774](https://doi.org/10.21105/joss.01774)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats on your new paper @munkm!!! Thanks very much to @lpantano for editing and to reviewers @harpolea and @KayleighRutherford — this couldn't happen without your time and expertise.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01774/status.svg)](https://doi.org/10.21105/joss.01774)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01774">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01774/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01774/status.svg
   :target: https://doi.org/10.21105/joss.01774
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01774/status.svg)](https://doi.org/10.21105/joss.01774)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
